src/odemis/acq/path.py

```diff
@@ -985,3 +1012,64 @@ def _waitTemperatureReached(self, comp, timeout=None):
         raise TimeoutError("Target temperature (%g C) not reached after %g s" %
                            (comp.targetTemperature.value, timeout))
 
+
+class _MimasOpticalPathManager(OpticalPathManager):
+    """
+    Special class for the MIMAS
+    """
+    def __init__(self, microscope):
+        self.microscope = microscope
+        if microscope.role != "mimas":
+            raise NotImplementedError("Microscope role '%s' unsupported" % (microscope.role,))
+        self._modes = copy.deepcopy(MIMAS_MODES)
+
+        self._chamber_view_own_focus = False  # To keep compatibility with __del__()
+
+        # To keep compatibility with the OpticalPathManager.setAcqQuality
+        self.quality = ACQ_QUALITY_FAST
+
+        # Initialize the information for guessMode()
+        self._role_to_mode = self._compute_role_to_mode(self._modes)
+
+        # keep list of all components, to avoid creating new proxies
+        # every time the mode changes
+        self._cached_components = model.getComponents()
+
+        # will take care of executing setPath asynchronously
+        self._executor = OneTaskExecutor()
+
+    def _doSetPath(self, path, detector):
+
+        if isinstance(path, stream.Stream):
+            if detector is not None:
+                raise ValueError("Not possible to specify both a stream, and a detector")
+            try:
+                mode = self.guessMode(path)
+            except LookupError:
+                logging.debug("%s doesn't require optical path change", path)
+                return
+            target = self.getStreamDetector(path)  # target detector
+        else:
+            mode = path
+            if mode not in self._modes:
+                raise ValueError("Mode '%s' does not exist" % (mode,))
+            comp_role = self._modes[mode][0]
+            if detector is None:
+                target = self._getComponent(comp_role)
+            else:
+                target = detector
+
+        logging.debug("Going to optical path '%s', with target detector %s.", mode, target.name)
+
+        # TODO: only accept moving iif the current position is "IMAGING" FM_IMAGING, or MILLING, which means the stage is within ACTIVE_RANGE
+        stage = self._getComponent("stage")
+        aligner = self._getComponent("align")
+        current_pos = move.getCurrentPositionLabel(stage.position.value, stage, aligner)
+        if current_pos not in (IMAGING, FM_IMAGING, MILLING):
+            logging.warning("Optical cannot be changed while in position %s", POSITION_NAMES[current_pos])
```

Comment:
This sentence sounds a bit strange, shouldn't it be "optical path cannot be changed ..." or "objective cannot be changed ..."?